spec/runtime/loader.spec.ts

```diff
@@ -0,0 +1,216 @@
+    const module = {
+      fn1: annotatedFn(HTTP_ENDPOINT),
+      g1: {
+        fn2: annotatedFn(EVENT_ENDPOINT),
+      },
+    };
+
+    const endpoints: Record<string, ManifestEndpoint> = {};
+    loader.extractEndpoints(module, endpoints);
+    expect(endpoints).to.be.deep.equal({
+      fn1: {
+        entryPoint: 'fn1',
+        ...HTTP_ENDPOINT,
+      },
+      'g1-fn2': {
+        entryPoint: 'g1.fn2',
+        ...EVENT_ENDPOINT,
+      },
+    });
+  });
+});
+
+describe('loadBackend', () => {
+  const expected: ManifestBackend = {
+    endpoints: {
+      v1http: {
+        platform: 'gcfv1',
+        entryPoint: 'v1http',
+        httpsTrigger: {},
+      },
+      v1callable: {
+        platform: 'gcfv1',
+        entryPoint: 'v1callable',
+        labels: {},
+        callableTrigger: {},
+      },
+    },
+    requiredAPIs: {},
+    specVersion: 'v1alpha1',
+  };
+
+  type Testcase = {
+    name: string;
+    modulePath: string;
+    expected: ManifestBackend;
+  };
+
+  function runTests(tc: Testcase) {
+    it('loads backend given relative path', async () => {
+      await expect(loader.loadBackend(tc.modulePath)).to.eventually.deep.equal(
+        tc.expected
+      );
+    });
+
+    it('loads backend given absolute path', async () => {
+      await expect(
+        loader.loadBackend(path.join(process.cwd(), tc.modulePath))
+      ).to.eventually.deep.equal(tc.expected);
+    });
+  }
+
+  let prev;
+  beforeEach(() => {
+    // TODO: When __trigger annotation is removed and GCLOUD_PROJECT is not required at runtime, remove this.
+    prev = process.env.GCLOUD_PROJECT;
+    process.env.GCLOUD_PROJECT = 'test-project';
+  });
+
+  afterEach(() => {
+    process.env.GCLOUD_PROJECT = prev;
+  });
+
+  describe('commonjs', () => {
+    const testcases: Testcase[] = [
+      {
+        name: 'basic',
+        modulePath: './spec/fixtures/sources/commonjs',
+        expected,
+      },
+      {
+        name: 'has main',
+        modulePath: './spec/fixtures/sources/commonjs-main',
+        expected,
+      },
+      {
+        name: 'grouped',
+        modulePath: './spec/fixtures/sources/commonjs-grouped',
+        expected: {
+          endpoints: {
+            v1http: {
+              platform: 'gcfv1',
+              entryPoint: 'v1http',
+              httpsTrigger: {},
+            },
+            'g1-v1callable': {
+              platform: 'gcfv1',
+              entryPoint: 'g1.v1callable',
+              labels: {},
+              callableTrigger: {},
+            },
+          },
+          requiredAPIs: {},
+          specVersion: 'v1alpha1',
+        },
+      },
+      {
+        name: 'scheduled',
+        modulePath: './spec/fixtures/sources/commonjs-scheduled',
+        expected: {
+          endpoints: {
+            scheduled: {
+              platform: 'gcfv1',
+              entryPoint: 'scheduled',
+              labels: {
+                'deployment-scheduled': 'true',
```

Comment:
I don't think we want to require this in the manifest. We'll just add it in the deployment process as necessary (that way we keep authoring a stack.yaml easy)

Reply:
Done!